### PR TITLE
[WIP] Coarse Insights documentation

### PR DESCRIPTION
**Pull request by j949wang**
WIP for coarse insights

### CODE REVIEW

File: docs/regions.mdx [hunk 1]
```diff
@@ -78,3 +78,99 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+First, [sign up](https://radar.com/signup) for Radar and get an API key.
+
+Contact your customer success manager to enable Coarse Insights.
+
+From there, [integrate the SDK](/documentation/sdk) and call `Radar.trackOnce()` or `Radar.startTracking()`, depending on your use case. Radar will set the home regions based on the location tracked, as well as update the recent home regions list with a new entry.
+
+## How it works
+
+The first time a user is tracked either in the day or upon creation, the determined regions are added to the list of recent home regions. The most frequently occurring entry is then set as the user's home regions.
```
Comment (david-goodfellow) [inline docs/regions.mdx:169]: let's include that the list makes up the past 30 days where at least on track call happened for the respective user. Or something like that

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "@mdx-js/react": "^1.6.21",
     "@svgr/webpack": "^5.5.0",
     "clsx": "^1.1.1",
+    "cspell": "^6.31.2",
```
Comment (david-goodfellow) [inline package.json:25]: what is this

File: docs/regions.mdx [hunk 3]
```diff
@@ -78,3 +78,99 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+As well as the following event context:
+```json
+{
+  "traveling": {
+    "country": true;
+    "dma": true;
+    "state": true;
+    "postalCode": true;
+    "city": true;
```
Comment (david-goodfellow) [inline docs/regions.mdx:154]: all should be `,` not `;`

File: docs/regions.mdx [hunk 4]
```diff
@@ -78,3 +78,99 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+      "stateRegion": "64c081605282e60012b10709",
+      "city": "64be9f7997d7f50050922a3b",
+    },
+    {
+      "countryRegion": "5cf694f66da6a800683f4d71",
+      "stateRegion": "5cf695086da6a800683f4e69",
+      "dma": "5cf694fb6da6a800683f4d92",
+      "postalCode": "5cf6955c6da6a800683f6753",
+    },
+  ]
```
Comment (david-goodfellow) [inline docs/regions.mdx:108]: this block in unimportant as it is an internal implementation detail

File: docs/regions.mdx [hunk 5]
```diff
@@ -78,3 +78,99 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+  "homeRegions": {
+    "countryRegion": "5cf694f66da6a800683f4d71",
+    "stateRegion": "5cf695086da6a800683f4e69",
+    "dma": "5cf694fb6da6a800683f4d92",
+    "postalCode": "5cf6955c6da6a800683f6753",
```
Comment (david-goodfellow) [inline docs/regions.mdx:113]: this bug was fixed in your pr this morning

File: docs/regions.mdx [hunk 6]
```diff
@@ -78,3 +78,99 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
 ## Postal codes
 
 Regions supports postal (ZIP) code detection in the US only. Each postal code has a unique 5-letter `code`.
+
+## Coarse Insights
+
+With Coarse Insights, you can leverage home region and traveling detection.
```
Comment (david-goodfellow) [inline docs/regions.mdx:84]: we should mention how this is globally around the world. Including a city field for non-US countries

File: package.json (hunk 2 above)
Comment (j949wang) [reply package.json:25]: The pre-commit linter complained not having this. Just does spell checking

Comment (kochis) [reply package.json:25]: oh good catch, I forgot to add it to the `package.json`

File: docs/regions.mdx (hunk 5 above)
Comment (j949wang) [reply docs/regions.mdx:95]: So I scrubbed this off the `webhookUserPayload` which is used for events. But on the `apiUserPayload` this still exists which is the callback on upsert. I think this is what `user` context is referring to?

File: package.json (hunk 2 above)
Comment (j949wang) [reply package.json:25]: Even the linter cannot block you ‼️

File: docs/regions.mdx [hunk 7]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
 ## Postal codes
 
 Regions supports postal (ZIP) code detection in the US only. Each postal code has a unique 5-letter `code`.
+
+## Coarse Insights
```
Comment (jsani-radar) [inline docs/regions.mdx:82]: nit: "Coarse insights" - we generally use sentence case in all copy

File: docs/regions.mdx [hunk 8]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+With Coarse Insights, you can leverage home region and traveling detection on global regions. This includes a `city` field for non-US countries.
```
Comment (jsani-radar) [inline docs/regions.mdx:84]: I'd remove the last sentence - `city` will be available in US soon, and we don't have cities for every country just yet

File: docs/regions.mdx [hunk 9]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+    "country": true,
+    "dma": true,
+    "state": true,
+    "postalCode": true,
+    "city": true,
+  }
+}
+```
+
+## Quickstart
```
Comment (jsani-radar) [inline docs/regions.mdx:141]: remove this "Quickstart" section as a whole - it's redundant to the one at the top of this page

File: docs/regions.mdx [hunk 10]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+First, [sign up](https://radar.com/signup) for Radar and get an API key.
+
+Contact your customer success manager to enable Coarse Insights.
```
Comment (jsani-radar) [inline docs/regions.mdx:145]: this line about contacting CS should be at the top of the Coarse insights section

File: docs/regions.mdx [hunk 11]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+From there, [integrate the SDK](/documentation/sdk) and call `Radar.trackOnce()` or `Radar.startTracking()`, depending on your use case. Radar will set the home regions based on the location tracked, as well as update the recent home regions list with a new entry.
+
+## How it works
```
Comment (jsani-radar) [inline docs/regions.mdx:149]: nit: `###` - this should be a subheader under Coarse insights

File: docs/regions.mdx [hunk 12]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+The first time a user is tracked either in the day or upon creation, the determined regions are added to the list of recent home regions. This list comprises the past 30 days in which at least one track call was made for the respective user. The most frequently occurring entry is then set as the user's home regions.
+If the most recent entry is different from the user's home regions, the user is determined to be traveling and the event context is updated accordingly.
+
+Home Regions can be enabled even if [Regions](/documentation/regions) event types are not.
```
Comment (jsani-radar) [inline docs/regions.mdx:154]: this last sentence isn't necessary - user state for all context types is available if event types are toggled off

File: docs/regions.mdx [hunk 13]
```diff
@@ -78,3 +78,81 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+First, [sign up](https://radar.com/signup) for Radar and get an API key.
+
+Contact your customer success manager to enable Coarse Insights.
+
+From there, [integrate the SDK](/documentation/sdk) and call `Radar.trackOnce()` or `Radar.startTracking()`, depending on your use case. Radar will set the home regions based on the location tracked, as well as update the recent home regions list with a new entry.
+
+## How it works
+
+The first time a user is tracked either in the day or upon creation, the determined regions are added to the list of recent home regions. This list comprises the past 30 days in which at least one track call was made for the respective user. The most frequently occurring entry is then set as the user's home regions.
```
Comment (jsani-radar) [inline docs/regions.mdx:151]: remove "either in the day or upon creation"

**Review by jsani-radar — COMMENTED** [review]
small comments for some general cleanup, nothing major

overall lgtm - feel free to merge upon approval

File: docs/regions.mdx (hunk 8 above)
Comment (j949wang) [reply docs/regions.mdx:84]: @david-goodfellow

File: docs/regions.mdx (hunk 10 above)
Comment (j949wang) [reply docs/regions.mdx:145]: Changed to "To enabled, contact yoru customer success manager"

File: docs/regions.mdx [hunk 14]
```diff
@@ -78,3 +78,71 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+      "type": "postalCode",
+      "name": "21014",
+      "code": "21014",
+      "icon": "🇺🇸",
+    }
+  }
+}
+```
+
+As well as the following event context:
```
Comment (palmakkar) [inline docs/regions.mdx:128]: nit: this should probably be a standalone sentence, something like "It also provides the following event context:"

File: docs/regions.mdx [hunk 15]
```diff
@@ -78,3 +78,71 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
 ## Postal codes
 
 Regions supports postal (ZIP) code detection in the US only. Each postal code has a unique 5-letter `code`.
+
+## Coarse insights
+
+With Coarse Insights, you can leverage home region and traveling detection on global regions. To enable, contact your customer success manager.
```
Comment (palmakkar) [inline docs/regions.mdx:84]: a little unclear from this that coarse insights are tied to a user - I'd suggest something like "leverage a user's home region..."

File: docs/regions.mdx [hunk 16]
```diff
@@ -78,3 +78,71 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+    "state": true,
+    "postalCode": true,
+    "city": true,
+  }
+}
+```
+
+### How it works
+
+The first time a user is tracked, the determined regions are added to the list of recent home regions. This list comprises the past 30 days in which at least one track call was made for the respective user. The most frequently occurring entry is then set as the user's home regions.
```
Comment (palmakkar) [inline docs/regions.mdx:143]: wasn't immediately clear to me what "recent home regions" before learning more about the implementation - wonder if we can explain the mechanism with something slightly more high level, like

"A user's home regions are determined by calculating which regions the user was most frequently tracked in within the last 30 days. If the regions from user's most recent location are different from the user's home regions, the user is determined to be traveling and the event context is updated accordingly."

**Review by palmakkar — COMMENTED** [review]
looks mostly good to me, couple of nits and a thought on the how it works blurb

File: docs/regions.mdx (hunk 15 above)
Comment (j949wang) [reply docs/regions.mdx:84]: would it make more sense to say something like, "You can leverage our Regions coverage to determine a user's home regions and detect traveling"

File: docs/regions.mdx (hunk 5 above)
Comment (david-goodfellow) [reply docs/regions.mdx:95]: per thread [here](https://radarlabs.slack.com/archives/C057UTBTF3L/p1690833597405579), lets remove these `<TYPE>Region` fields from the response. Might make sense to pull this into a helper function since we already do it for `webhookUserPayload`

File: docs/regions.mdx [hunk 17]
```diff
@@ -78,3 +78,67 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+## Coarse insights
+
+With Coarse Insights, you can determine a user's home regions and detect traveling across our global coverage. To enable, contact your customer success manager.
+
+Coarse Insights provides the following user context:
+
+```json
+{
+  "recentHomeRegions": [],
```
[the remark this answers is not in the document]
Comment (j949wang) [reply docs/regions.mdx:90]: Sounds good

File: docs/regions.mdx [hunk 18]
```diff
@@ -78,3 +78,67 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
...
+
+It also provides the following event context:
+```json
+{
+  "traveling": {
+    "country": true,
+    "dma": true,
+    "state": true,
+    "postalCode": true,
+    "city": true,
```
Comment (david-goodfellow) [inline docs/regions.mdx:131]: I think we should not list `city` alongside `dma` here to prevent confusion. Let's just stick with `dma` here since the above response does not have city

File: docs/regions.mdx [hunk 19]
```diff
@@ -78,3 +78,65 @@ Regions supports DMA (designated market area) detection in the US only. Each DMA
 ## Postal codes
 
 Regions supports postal (ZIP) code detection in the US only. Each postal code has a unique 5-letter `code`.
+
+## Coarse insights
+
+With Coarse Insights, you can determine a user's home regions and detect traveling across our global coverage. To enable, contact your customer success manager.
```
Comment (palmakkar) [inline docs/regions.mdx:84]: nit: "across our global coverage" sounds a bit wordy, can it be replaced with "globally"?

**Review by palmakkar — APPROVED** [review]
looking good, added one final nit then good to merge from my end

File: docs/regions.mdx (hunk 19 above)
Comment (j949wang) [reply docs/regions.mdx:84]: Definitely
